### DISCUSSION TITLE
Confusing example in documentation

### DISCUSSION THREAD

**Discussion by imMatheus** [pmndrs/jotai#2395] [Bug report]
## Summary

In the [perfomance](https://jotai.org/docs/guides/performance) section of the docs, at the end there is a code example that seems to be outdated. If you look at the code for the `Item` component, in the use useMemo its referring to a `val` variable in the  `useAtom(atom(val))` part. But this variable does not exist in the rest of the code. Im guessing that it should be `initialValue`. Also its pretty weird that the `Item` code has `React.useMemo` but the `List` just uses `useMemo`, In my opinion this can be confusing as to why there is a difference. 

![Screenshot 2024-02-12 at 13 27 34](https://github.com/pmndrs/jotai/assets/77362975/abb5288e-9160-43ab-b61a-dd02201655a4)

## Link to reproduction

[https://jotai.org/docs/guides/performance](https://jotai.org/docs/guides/performance)

## Check List

Please do not ask questions in issues.

- [x] I've already opened a [discussion](https://github.com/pmndrs/jotai/discussions) before opening this issue, or already discussed in other media.

Please include a minimal reproduction.

- [x] I've added a link to a [typescript playground](https://www.typescriptlang.org/play) or [codesandbox](https://codesandbox.io) with a minimal reproduction.

**Comment by dai-shi** [comment]
Can you please open a PR to improve docs?

**Comment by dai-shi** [comment]
Or, feel free to ask questions here before working on it.

**Reply by math-krish** [reply]
I tried implementing the example given in the documentation and I get the following error
`React Hook "useAtom" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function.eslint[react-hooks/rules-of-hooks](https://reactjs.org/docs/hooks-rules.html)`
I can try taking this up but with little more context on what is trying to happen, could you answer the following questions?
1. If we are using a local state `setToggle` then why are we trying to update the parent as well? Could you explain the usecase of what the problem statement itself is?  
2.  Do we actually require a `useMemo` at all? 
I'm pretty new to jotai, would be great if you could help

**Reply by dai-shi** [reply]
Thanks for the heads-up. This guide is not well written at all. "Memoized pattern" section should be removed as it's just wrong. Or, I'm fine to remove this entire "Performance" guide.

**Reply by math-krish** [reply]
I can take it up, could you assign it to me?
I think performance guide has aspects about the `focusAtom` and `splitAtom` which might be useful as it can be combined with the useMemo (found the following example in your code: 
``` 
  const nameAtom = useMemo(
    () => focusAtom(fieldAtom, (o) => o.prop('name')),
    [fieldAtom],
  )

```
Maybe we can remove everything from the ["Stop observing" pattern](https://jotai.org/docs/guides/performance#stop-observing-pattern) section.

**Reply by dai-shi** [reply]
Okay, please open a draft PR and let's discuss details there. (that implies "assigned.")

**Reply by math-krish** [reply]
[PR Link](https://github.com/pmndrs/jotai/pull/2712) 
@dai-shi Let me know if there is anything else required. Thank you for letting me contribute!
